app/views/Settings/LocationTrackingStatus.js

```diff
@@ -0,0 +1,184 @@
+
+  const isGPS = config.tracingStrategy === 'gps';
+
+  const hasLocTrackingPermissions =
+    reason === Reason.USER_ENABLED || reason === Reason.USER_DISABLED;
+
+  const isMissingLocPermissions =
+    reason === Reason.DEVICE_LOCATION_OFF ||
+    reason === Reason.APP_NOT_AUTHORIZED;
+
+  const getCurrentRowDirection = i18n.dir() === 'rtl' ? 'row-reverse' : 'row';
+
+  const getLocStatusElement = () => {
+    console.log(`reason: ${reason}`);
+    console.log(
+      `hasLocTrackingPermissions: ${hasLocTrackingPermissions} isMissingLocPermissions: ${isMissingLocPermissions}`,
+    );
+
+    if (hasLocTrackingPermissions) {
+      return (
+        <LocationStatusSwitch
+          canTrack={canTrack}
+          setLocTrackingStatus={setLocTrackingStatus}
+        />
+      );
+    }
+
+    if (isMissingLocPermissions) {
+      return <EnableLocationButton reason={reason} />;
+    }
+  };
+
+  if (isGPS) {
+    return (
+      <>
+        <Container
+          style={css`
+            flex-direction: ${getCurrentRowDirection};
+          `}>
+          {getLocStatusElement()}
+        </Container>
+        <Divider />
+      </>
+    );
+  } else {
+    return null;
+  }
+};
+
+const EnableLocationButton = ({ reason }) => {
+  const { t } = useTranslation();
+
+  /**
+   * Conditional toggling based on the location permission actions
+   * that are enabled per platform.
+   */
+  const requestLocationPermission = async () => {
+    /**
+     * Open the settings for the app
+     */
+    if (reason === Reason.APP_NOT_AUTHORIZED) {
+      openSettings();
+    }
+
+    /**
+     * Open Device Settings.
+     * If iOS, open Settings home screen
+     * If Android, we are able to open the device location permissions
+     */
+    if (reason === Reason.DEVICE_LOCATION_OFF) {
+      if (isPlatformiOS()) {
+        const deviceSettingsUrl = 'App-prefs:';
+        Linking.openURL(deviceSettingsUrl);
+      } else {
+        DeviceSettings.location();
+      }
+    }
+  };
+
+  return (
+    <>
+      <Button
+        onPress={requestLocationPermission}
+        label={t('label.logging_request_location_permission')}
+        style={{ width: '100%' }}
+        secondary
+        small
+      />
+    </>
+  );
+};
+
+const LocationStatusSwitch = ({ canTrack, setLocTrackingStatus }) => {
+  console.log('LocationStatusSwitch!');
+  const { t, i18n } = useTranslation();
+  const [label, setLabel] = useState('');
+
+  useEffect(() => {
+    let labelInterval;
+
+    /**
+     * Provide a basic animation to show that location logging
+     * is active. If the label reaches three periods, reset
+     * to the original label
+     */
+    const createLabelInterval = () => {
+      return setInterval(() => {
+        setLabel(label => {
+          if (label.slice(-3) === '...') {
+            setLabel(t('label.logging_active_location'));
+          } else {
+            setLabel(label + '.');
+          }
+        });
+      }, 1000);
```

Comment:
Perhaps the interval can be configured when it's a hook.

Reply:
Done